### PR TITLE
stylo: Use master bindgen again.

### PR DESCRIPTION
**Pull request by emilio**
To be fair, this is master bindgen with https://github.com/servo/rust-bindgen/pull/318.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14460)

### CODE REVIEW

**Comment by emilio** [comment]
r? @heycam 

cc @upsuper

**Comment by emilio** [comment]
Main changes are moving all the generated types to `structs`, and leave `bindings.rs` only with functions. The rest is only naming / namespace changes.

File: ports/geckolib/glue.rs [hunk 1]
```diff
@@ -607,7 +606,7 @@ pub extern "C" fn Servo_DeclarationBlock_GetCssText(declarations: RawServoDeclar
 #[no_mangle]
 pub extern "C" fn Servo_DeclarationBlock_SerializeOneValue(
     declarations: RawServoDeclarationBlockBorrowed,
-    property: *mut nsIAtom, is_custom: bool,
+    property: *const nsIAtom, is_custom: bool,
```
Comment (heycam) [inline ports/geckolib/glue.rs]: Could we leave this change and the get_property_name_from_atom change out of this patch?  We'll have to undo them once https://bugzilla.mozilla.org/show_bug.cgi?id=1322098 lands anyway.

**Comment by heycam** [comment]
r=me besides that.

**Comment by emilio** [comment]
Fair, just ran it with your patch locally applied.

@bors-servo r=heycam

**Comment by emilio** [comment]
@bors-servo: retry

 * filter-intermittents?

**Comment by KiChjang** [comment]
@emilio [This](http://build.servo.org/builders/linux-rel-wpt/builds/1299/steps/shell__1/logs/filtered-wpt-errorsummary.log) means that there is a new intermittent.

**Comment by KiChjang** [comment]
@bors-servo retry
* #14462

**Pull request merged**